contracts/upgradeable_contracts/amb_erc677_to_erc677/ForeignAMBErc677ToErc677RelativeDailyLimit.sol

```diff
@@ -0,0 +1,54 @@
+pragma solidity 0.4.24;
+
+import "./ForeignAMBErc677ToErc677.sol";
+import "../RelativeExecutionDailyLimit.sol";
+
+contract ForeignAMBErc677ToErc677RelativeDailyLimit is ForeignAMBErc677ToErc677, RelativeExecutionDailyLimit {
+    function handleBridgedTokens(
+        address _recipient,
+        uint256 _value,
+        bytes32 _nonce
+    ) public {
+        _updateTodayLimit();
+        super.handleBridgedTokens(_recipient, _value, _nonce);
```

Comment:
Do we need to call `_updateTodayLimit`? It seems that this method calls `_calculateLimit()` the first time when the limit is not set.
Then inside `handleBridgedTokens`, the method `withinExecutionLimit` will be called, then `executionDailyLimit`,  and then `_getTodayLimit()` which will try to call `_calculateLimit()` if it wasn't calculated.

Unless I'm missing something I think we could remove `_updateTodayLimit` and its calls so only `_getTodayLimit` is called and update `_getTodayLimit` to something like this:
```
function _getTodayLimit() internal returns (uint256) {
    uint256 limit = _todayLimit();
    if (limit == 0) {
        // not set yet
        limit = _calculateLimit();
        _setTodayLimit(limit);
        emit TodayLimitSet(limit);
    }
    return limit;
}
```
This same behavior could be applied for all other bridge modes where relative daily limits are used too.
What do you think?

Reply:
Good catch, @patitonar!
I thought about it. But if we call `_setTodayLimt` inside `_getTodayLimit` then `executionDailyLimit` and `dailyLimit` will be not `view` methods

Reply:
You're right, I missed that. Thanks